src/main/python/hydra/lib/hydrabase.py

```diff
@@ -46,14 +46,15 @@ class AppGroup(object):
     group_name:     group name
     analyser:       Analyser class "name" e-g HAnalyser not HAnalyser()
     """
-    def __init__(self, hydra, app_name, group_name, analyser=None):
+    def __init__(self, hydra, app_name, group_name, num_app_instances, analyser=None):
         self.group_info = {}
         self.hydra = hydra
         if not analyser:
             raise Exception("AppGroup needs analyser class name passed as a name, curr val = %s" % analyser)
         self.analyser = analyser
         self.group_name = group_name
         self.app_name = app_name
+        self.num_app_instances = num_app_instances
```

Comment:
It is being used by library user (for example, kraken here). If you see analyze_write_sub_results() of https://gerrit.plumgrid.com:8443/#/c/23318/8/kraken/plumgrid/tests/pg_kraken_sbk.py , you will see that we use this to make sure that number of returned results of subscribers are same as  number of subscriber's app instances group. 
Just compare it with analyze_sub_results() of pubsub test case to get further idea. When you have multiple subscriber groups, this thing will help you by immediately telling you that how many results you should expect in a group.